Framework/Algorithms/src/PolarizationCorrectionWildes.cpp

```diff
@@ -528,6 +543,34 @@ API::WorkspaceGroup_sptr PolarizationCorrectionWildes::groupOutput(const Workspa
   return outWS;
 }
 
+/**
+ * Add an output name in the correct position in the vector and to the ADS.
+ * @param names A list of the names of the workspaces the algorithm has generated.
+ * @param spinStateOrder The order the output should be in.
+ * @param baseName The base name for the output workspaces ("BASENAME_SPINSTATE" e.g "OUTNAME_+-")
+ * @param ws The workspace to add to the vector and ADS.
+ * @param spinState The spin state the workspace represents.
+ */
+void PolarizationCorrectionWildes::addSpinStateOutput(std::vector<std::string> &names,
+                                                      const std::string &spinStateOrder, const std::string &baseName,
+                                                      const API::MatrixWorkspace_sptr &ws,
+                                                      const std::string &spinState) {
+  if (spinStateOrder.empty()) {
+    names.emplace_back(baseName + "_" + spinState);
+    API::AnalysisDataService::Instance().addOrReplace(names.back(), ws);
+  } else {
+    const auto &maybeIndex = PolarizationCorrectionsHelpers::indexOfWorkspaceForSpinState(
+        PolarizationCorrectionsHelpers::splitSpinStateString(spinStateOrder), spinState);
+    if (!maybeIndex.has_value()) {
+      throw std::invalid_argument("Required spin state (" + spinState + ") not found in spin state order (" +
```

Comment:
Not necessary to change if you'd prefer not to as this validation does work to prevent invalid spin states being used and provides a clear error message. I'm just conscious that we only get to this once the full correction has been applied. For example, if I run the algorithm with spin state "-, +" then it's only at this final stage that we pick up that it's not a valid input, however I think we could potentially validate for this earlier.

Reply:
Am I right in thinking that we could fix this by only allowing the two char spin states? That's easy to set on the validator rather than doing heavier checking.